### PR TITLE
[ Fix ] Service Handler

### PR DESCRIPTION
**Pull request by bukandicki**
### Changelog/Description
Fix and update logic `try-catch` with `finally`, please check this reference why I used `finally` here. Anyway it causing error on the checkout page, caused `isLoading` not to be `false` if we do not assign a callback function

> ### **finally_statements**
> Statements that are executed after the try statement completes. These statements execute regardless of whether an exception was thrown or caught.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

### CODE REVIEW

File: src/mixins/serviceHandler.js [hunk 1]
```diff
@@ -55,6 +57,8 @@ export default {
         console.error(`[${error?.response?.status}] Something went wrong, Please try again later!`)
 
         return Promise.reject(error)
+      } finally {
+        this.isLoading = false
```
Comment (agustinustheo) [inline src/mixins/serviceHandler.js:61]: This is not how you should handle `isLoading` using Polkadot.

I assume you are doing this because your cancel order loading button is not processing correctly, you shouldn't set `isLoading` to false inside a `finally` because it will resolve before events are fulfilled by the blockchain. You should set `isLoading` inside the callback function I made.

Your code will break other functionalities in production. For example, I ran your code in my development environment and the loading symbol no longer waits until the event finishes in the blockchain.

Please refer to this commit: https://github.com/debionetwork/debio-frontend/commit/17dbe1d0e67b1de941a08e182f893648acf4d0d5

Please read this doc of PolkadotJS: https://polkadot.js.org/docs/api/start/api.query.subs/#subscriptions-with-params

And finally, please remove this. If you have questions or other proposals please reach me via Discord.

**Review by agustinustheo — CHANGES_REQUESTED** [review]
Please remove the `finally` block.

TLDR; it's not about `finally`, you should subscribe to the PolkadotJS API by adding a callback and set `isLoading` from there.

File: src/mixins/serviceHandler.js (hunk 1 above)
Comment (agustinustheo) [reply src/mixins/serviceHandler.js:61]: Oh and if you're wondering WHY cancel order callback doesn't work, it's because the callback hasn't been set inside the `polkadotProvider`s please check the difference between these two files:
- https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/orders.js
- https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/labs/index.js

These two files use PolkadotJS, but all `labs` commands use a callback while most `orders` commands do not. Anyway, you can even check with the `fulfillOrder` command inside `orders` here: https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/orders.js#L8

Comment (agustinustheo) [reply src/mixins/serviceHandler.js:61]: The solution? Make a separate universal callback function similar to the one you see here: https://github.com/debionetwork/debio-frontend/blob/development/src/lib/polkadotProvider/command/labs/index.js#L27

Does it seem repetitive? Yes, it does. You are open to suggest inputs if you like. But so far, this is how you should do it.

Comment (bukandicki) [reply src/mixins/serviceHandler.js:61]: Hi @agustinustheo, I have a question. Does that mean all requests have to use callbacks?

Comment (agustinustheo) [reply src/mixins/serviceHandler.js:61]: All commands must use callbacks @bukandicki, albeit not all of them will use them for the same reason.h

**Comment by bukandicki** [comment]
Move to another branch
